pyopencl/array.py

```diff
@@ -2579,6 +2579,38 @@ def hstack(arrays, queue=None):
 
     return result
 
+
+def stack(arrays, axis=0, queue=None):
+    if not arrays:
+        raise ValueError("need at least one array to stack")
+
+    if axis != 0:
+        # pyopencl.Array.__setitem__ does not support non-contiguous assignments
+        raise NotImplementedError("axis!=0 not implemented")
```

Comment:
You can easily generalize this to Fortran-layout arrays `ary.flags.f_contiguous` by transposing inputs and outputs. (should test this as well)

Reply:
Yep, works for f_contiguous arrays too: ad6b6957a0fef0e65b55660311d5830a2c72e8b5